### DISCUSSION TITLE
Problem with geometry/geography fields

### DISCUSSION THREAD

**Discussion by sdekna** [hasura/graphql-engine#9095] [Question]
I am trying to query a `Sellers` table `where` the location of the seller is within `x` m from location provided.
This is how my query looks like:
```
query sellersInRange ($location: geography! ) {
  Sellers(where: {location: {_cast: {geography: {_st_d_within: {distance: 20000, from: $location}}}}}) {
    id
    status
    description
    is_new
    location
    image
    name
    Products{
      Product{
        id
        name
        price
        description
        Sauces{
          Sauce{
            id
            name
            image
          }
        }
      }
    }
  }
}
```
The `location` variable is: `{"location": {"type": "Point" ,"coordinates": [13.191394465831182,32.88723208673258]}}`.
Using hasura's graphiql, when I write this variable hasura gives me a type error saying "Expecting value of type geography", when it is already so. If I make the query hasura, it gives me the desired results.

My problems lie here:
1- The field is of type `geometry` not `geography` (as hasura demands), and if I change the field type to `geography`, then hasura demands `geography`.
2- The query runs ok in graphiql but if I call it using any graphql client, it fails and gives me a "query error"... other queries work ok but when I attempt for the location querying it fails outside of graphiql.

Any ideas?

**Comment by adas98012** [comment]
I would like to share an old blog with you, which seems quite relevant to your usage of Hasura GraphQL Engine:

https://hasura.io/blog/graphql-and-geo-location-on-postgres-using-hasura-562e7bd47a2f/

The blog is a bit dated, but it may help you get unblocked. 
We would like to hear back from you about your specific outcome.

**Reply by sdekna** [reply]
Many thanks for the reply, however, it still doesn't solve the issue, because it seems to be an issue with Hasura... `geometry` and `geaography` seem to switch themselves... if I edit the field to be a $geometry field, the graphql input requires it to be `geography`, if I edit the field back to `geometry` graphql input requires `geography`, and then if I give it what it wants, I get a warning saying `expected value of type geography` and vice versa.

I believe it could be a bug.

**Comment by Knorway** [comment]
I'm facing the same issue and this is currently working outside of hasura graphiQL on both apollo client and axios for me.
```
const { data } = useQuery(
		gql`
			query MyQuery($from: geometry!) {
				location(
					where: { point: { _st_d_within: { distance: 1.5, from: $from } } }
				) {
					id
					point
				}
			}
		`,
		{
			variables: {
				from: {
					type: 'Point',
					coordinates: [-49, 20],
				},
			},
		}
	);
```
but I'm still not sure querying geometry type data ignoring input type error is the best I can get using hasura
![스크린샷 2022-10-14 오후 6 04 02](https://user-images.githubusercontent.com/66667442/195830143-ad37e0b4-1942-45dc-9863-36fa00c832e3.png)

edit: this could be a workaround for simple cases at the moment. thanks to [gegtik at discord](https://discord.com/channels/407792526867693568/535727661167673364/1030586294193094726)
![스크린샷 2022-10-15 오후 1 06 10](https://user-images.githubusercontent.com/66667442/195968111-f109640f-97f7-4715-8766-883ce6679cf0.png)

**Comment by adas98012** [comment]
@manasag This input error surfaces on the console, but the query gets executed successfully.

**Reply by dsandip** [reply]
Converted this to an issue. Locking the discussion to take this forward in the issue.

@sdekna @Knorway Thanks for reporting this!
